src/test/java/rx/swing/sources/ChangeEventSourceTest.java

```diff
@@ -0,0 +1,491 @@
+                        .subscribe(testSubscriber);
+
+                testSubscriber.assertNoErrors();
+                testSubscriber.assertNoValues();
+
+                sub.unsubscribe();
+
+                testSubscriber.assertUnsubscribed();
+
+                tabbedPane.setSelectedIndex(2);
+
+                testSubscriber.assertNoErrors();
+                testSubscriber.assertNoValues();
+
+                assertEquals(numberOfListenersBefore, tabbedPane.getChangeListeners().length);
+            }
+        }).awaitTerminal();
+    }
+
+    @Test
+    public void fromChangeEventsOf_usingObjectWithoutExpectedChangeListenerSupport_failsFastWithException() throws Throwable {
+        SwingTestHelper.create().runInEventDispatchThread(new Action0() {
+
+            @Override
+            public void call() {
+                try {
+                    ChangeEventSource.fromChangeEventsOf("doesNotSupportChangeListeners").subscribe(TestSubscriber.create());
+                    fail(IllegalArgumentException.class.getSimpleName() + " expected");
+                } catch (IllegalArgumentException ex) {
+                    assertEquals("'doesNotSupportChangeListeners' has not the expected signature to support change listeners in rx.swing.sources.ChangeEventSource",
+                            ex.getMessage());
+                }
+
+                try {
+                    ChangeEventSource.fromChangeEventsOf(new Object() {
+                        private void addChangeListener(ChangeListener changeListener) {/* no-op */ }
+
+                        private void removeChangeListener(ChangeListener changeListener) {/* no-op */ }
+
+                        @Override
+                        public String toString() {
+                            return "hasWrongMethodModifiers";
+                        }
+                    }).subscribe(TestSubscriber.create());
+                    fail(IllegalArgumentException.class.getSimpleName() + " expected");
+                } catch (IllegalArgumentException ex) {
+                    assertEquals("'hasWrongMethodModifiers' has not the expected signature to support change listeners in rx.swing.sources.ChangeEventSource",
+                            ex.getMessage());
+                }
+            }
+        }).awaitTerminal();
+    }
+
+    @Test
+    public void issuesWithAddingChangeListenerOnSubscriptionArePropagatedAsError() throws Throwable {
+        SwingTestHelper.create().runInEventDispatchThread(new Action0() {
+
+            @Override
+            public void call() {
+                TestSubscriber<ChangeEvent> testSubscriber = TestSubscriber.create();
+
+                JProgressBar progressBarWithBrokenChangeListenerSupport = new JProgressBar() {
+                    @Override
+                    public void addChangeListener(ChangeListener listener) {
+                        if (listener.getClass().getName().contains(ChangeEventSource.class.getSimpleName())) {
```

Comment:
The addChangeListener method is also called by UI-classes like BasicProgressBarUI, that's the reason why it may not throw always. The idea here is to simulate some kind of JProgressBar subclass with a somehow broken implementation and check the propagation.